var/spack/repos/builtin/packages/steps/package.py

```diff
@@ -12,51 +14,127 @@ class Steps(CMakePackage):
+    )
     variant("lapack", default=False, description="Use new BDSystem/Lapack code for E-Field solver")
-    variant("petsc", default=False, description="Use PETSc library for parallel E-Field solver")
+    variant("distmesh", default=True, description="Add solvers based on distributed mesh")
+    variant("petsc", default=True, description="Use PETSc library for parallel E-Field solver")
     variant("mpi", default=True, description="Use MPI for parallel solvers")
+    variant("coverage", default=False, description="Enable code coverage")
+    variant("bundle", default=False, description="Use bundled libraries")
+    variant("stochtests", default=True, description="Add stochastic tests to ctests")
+    variant(
+        "build_type",
+        default="RelWithDebInfo",
+        description="CMake build type",
+        values=("Debug", "Release", "RelWithDebInfo", "MinSizeRel", "RelWithDebInfoAndAssert"),
+    )
+    variant(
+        "caliper", default=False, description="Build in caliper support (Instrumentor Interface)"
+    )
+    variant(
+        "likwid", default=False, description="Build in likwid support (Instrumentor Interface)"
+    )
 
+    # Build with `ninja` instead of `make`
+    generator = "Ninja"
+
+    conflicts("+distmesh~mpi", msg="steps+distmesh requires +mpi")
+    depends_on("benchmark", type=("build", "test"))
     depends_on("blas")
+    depends_on("boost", type="build")
+    depends_on("caliper", when="+caliper")
+    depends_on("easyloggingpp", when="~bundle")
+    depends_on("gmsh", when="+distmesh")
     depends_on("lapack", when="+lapack")
+    depends_on("lcov", when="+coverage", type="build")
+    depends_on("likwid", when="+likwid")
+    depends_on("metis+int64")
     depends_on("mpi", when="+mpi")
-    depends_on("petsc~debug+int64", when="+petsc")
-    depends_on("python")
+    depends_on("ninja", type="build")
+    depends_on("omega-h+gmsh+mpi", when="~bundle+distmesh")
+    depends_on("petsc~debug+int64+mpi", when="+petsc+mpi")
+    depends_on("petsc~debug+int64~mpi", when="+petsc~mpi")
+    depends_on("pkgconfig", type="build")
     depends_on("py-cython")
+    depends_on("py-gcovr", when="+coverage", type="build")
+    depends_on("py-h5py", type=("build", "test", "run"))
+    depends_on("py-matplotlib", type=("build", "test"))
+    depends_on("py-mpi4py", when="+distmesh", type=("build", "test", "run"))
+    depends_on("py-numpy", type=("build", "test", "run"))
+    depends_on("py-scipy", type=("build", "test", "run"))
+    depends_on("python", type=("build", "test", "run"))
+    depends_on("random123", when="~bundle")
+    depends_on("sundials@:2.99.99+int64", when="~bundle")
 
     patch("for_aarch64.patch", when="target=aarch64:")
 
+    def patch(self):
+        # easylogging requires compilation by
+        # its dependents: splice in disabling all errors
+        filter_file(r"(-Wno-double-promotion)", r"-Wno-error \1", "src/steps/util/CMakeLists.txt")
```

Comment:
Given the comment, is this only needed when `~bundle`?

Reply:
This is needed in both cases. This piece of dirt will be cleaned up in the next release.